rust/core-lib/src/file.rs

```diff
@@ -226,12 +227,11 @@ fn try_save(
 
     fs::rename(tmp_path, path)?;
 
-    if let Some(info_unwrapped) = file_info {
-        #[cfg(target_family = "unix")]
-        fs::set_permissions(
-            path,
-            Permissions::from_mode(info_unwrapped.permissions.unwrap_or(0o644)),
-        )?;
+    #[cfg(target_family = "unix")]
```

Comment:
is this necessary if we have the `#[allow(unused)]` annotation?

Reply:
I believe so, because [`from_mode`](https://doc.rust-lang.org/std/os/unix/fs/trait.PermissionsExt.html#tymethod.from_mode) is a function which only exists for unix systems, and the field [`permissions` on `FileInfo`](https://github.com/xi-editor/xi-editor/blob/533a66aa91b019fdb69b7b4c3a90fc38c9c5f59a/rust/core-lib/src/file.rs#L49) only exists on unix systems too? see:

Reply:
okay, just checking. Basically: it's nice to have small patches, and if the previous version of the code now doesn't produce warnings that's great, but if this is required that's good too!